uaparser/options.go

```diff
@@ -0,0 +1,45 @@
+package uaparser
+
+type Option func(*Parser)
+
+func WithMode(mode LookupMode) Option {
+	return func(s *Parser) {
+		s.config.Mode = mode
+	}
+}
+
+func WithSort(useSort bool) Option {
+	return func(s *Parser) {
+		s.config.UseSort = useSort
+	}
+}
+
+func WithDebug(debug bool) Option {
+	return func(s *Parser) {
+		s.config.DebugMode = debug
+	}
+}
+
+func WithCacheSize(size int) Option {
+	return func(s *Parser) {
+		s.config.CacheSize = size
+	}
+}
+
+func WithMissesThreshold(threshold uint64) Option {
+	return func(s *Parser) {
+		s.config.MissesThreshold = threshold
+	}
+}
```

Comment:
one thing I am pondering, this option is only meaningful in combination with UseSort, i.e. `New(WithMissesThreshold(...))` is no different than `New()` in behavior.  would it make sense for MissesThreshold to be a parameter to WithSort?